src/main/java/io/quarkus/search/app/hibernate/AnalysisConfigurer.java

```diff
@@ -29,7 +29,7 @@ public class AnalysisConfigurer implements ElasticsearchAnalysisConfigurer {
     // This is simplified by assuming no default package, lowercase package names and capitalized class name,
     // so we get fewer false positives
     private static final Pattern SIMPLIFIED_JAVA_CLASS_NAME_CAPTURE_PATTERN = Pattern
-            .compile("(?:[a-z_$][a-z0-9_$]*\\.)+([A-Z][A-Za-z0-9_$]*)$");
+            .compile("(?:[a-z_$][a-z0-9_$]*\\.)+([a-z][a-z0-9_$]*)$");
```

Comment:
This will result in configuration properties or domain names being mistaken for Java class names, and their last component being indexed as a separate token... This could lead to rather odd behavior.
Can't we move the "compoundTechnicalNameFilter" in very first position instead? Before the lowercase filter?